src/DeployPage.js

```diff
@@ -0,0 +1,635 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ky from 'ky';
+import TimeAgo from 'react-timeago';
+import classNames from 'classnames';
+
+import AutoProgressBar from './AutoProgressBar';
+import shortUrls from './shortUrls';
+
+const BORS_LOGIN = 'bors[bot]';
+
+export default class DeployPage extends React.Component {
+  static propsTypes = {
+    owner: PropTypes.string.isRequired,
+    repo: PropTypes.string.isRequired,
+    deployments: PropTypes.arrayOf(
+      PropTypes.shape({
+        url: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired
+      })
+    ).isRequired
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      commits: null,
+      deployInfo: null,
+      error: null,
+      loading: null,
+      tags: null
+    };
+  }
+
+  isLoading() {
+    if (this.state.loading === null) {
+      return true;
+    }
+    return this.state.loading.size > 0;
+  }
+
+  startLoad(name) {
+    this.setState(({ loading }) => {
+      if (!loading) {
+        loading = new Set();
+      }
+      loading.add(name);
+      return { loading };
+    });
+  }
+
+  finishLoad(name) {
+    this.setState(state => state.loading.delete(name));
+  }
+
+  async fetchShas() {
+    const { owner, repo, deployments } = this.props;
+    this.startLoad('shas');
+    try {
+      const res = await ky
+        .post('/shas', { json: { owner, repo, deployments } })
+        .json();
+
+      this.setState({
+        deployInfo: res.deployments,
+        tags: res.tags
+      });
+    } catch (error) {
+      this.setState({ error });
+    } finally {
+      this.finishLoad('shas');
+    }
+  }
+
+  async fetchCommits() {
+    const { owner, repo } = this.props;
+    this.startLoad('commits');
+    try {
+      const commitsUrl = new URL(window.location.origin);
+      commitsUrl.pathname = '/githubapi/commits';
+      commitsUrl.searchParams.set('owner', owner);
+      commitsUrl.searchParams.set('repo', repo);
+      commitsUrl.searchParams.set('per_page', 100);
+
+      const commits = await ky.get(commitsUrl).json();
+
+      this.setState({ commits });
+    } catch (error) {
+      this.setState({ error });
+    } finally {
+      this.finishLoad('commits');
+    }
+  }
+
+  componentDidMount() {
+    this.fetchShas();
+    this.fetchCommits();
+  }
+
+  render() {
+    const { owner, repo } = this.props;
+    const { error, loading, deployInfo, commits, tags } = this.state;
+
+    document.title = `What's deployed on ${owner}/${repo}?`;
```

Comment:
nit: "Deployed", not "deployed". Also, the trailing question marks is supposed to disappear based on it now being known. So the question mark should change depending on `this.isLoading()`.